sdk/go/common/itime/clock.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016-2024, Pulumi Corporation.
```

Comment:
Could also look at just using https://github.com/jonboulle/clockwork for this. It looks pretty decent.